redhat/rdma-core.spec

```diff
@@ -301,6 +315,9 @@ easy, object-oriented access to IB verbs.
          -DCMAKE_INSTALL_UDEV_RULESDIR:PATH=%{_udevrulesdir} \
          -DCMAKE_INSTALL_PERLDIR:PATH=%{perl_vendorlib} \
          -DENABLE_IBDIAGS_COMPAT:BOOL=True \
+%if ! %{with builddocs} 
+         -DNO_MAN_PAGES=1 \
```

Comment:
No, package build scripts can never include NO_MAN_PAGES - shipping rdma-core without man pages is completely unacceptable.

Reply:
What if we switched the bcondition at the top to always build docs by default? The problem I'm trying to solve here is to not require pandoc in downstream specs (which are already shipping prebuilt docs anyways), while allowing us to do test builds against the upstream spec too.

Reply:
The cmake automatically uses the prebuilt docs, there is no need to change the specfile at all.

Reply:
I'm trying to remove the BuildRequires on pandoc from the spec. So to ask my question another way: is there some way I can use this upstream spec to build a non-production build on an OS that does not ship pandoc?

Reply:
Remove the buildrequires and everything will work if your source tar has the correct pre-built man pages in it.

Reply:
> Remove the buildrequires and everything will work if your source tar has the correct pre-built man pages in it.

**_NO_**, this won't work for source tarball generated from upstream git repo. We **always** have to test patch against latest
upstream repo.

Reply:
For SUSE we use the git repo but pre-build the doc before submitting to our build system.
It adds a step but solved the pandoc dependency issue for us.

Reply:
@Honggang-LI if it doesn't work then the source tgz you are using doesn't have docs prebuilt and, again, complete NO to distros shipping without docs. You have to use tgz with prebuilt doc, and if you do then no changes to the spec file are required. See Nichola's note for how suse does it.

Reply:
@Honggang-LI The spec file and the gen-pandoc.sh script can be found here if you want to have a look:
https://build.opensuse.org/package/show/science:HPC/rdma-core